added proj_render() to checks which require compilation

ambertin opened the pull request:
For `has_no_absolute_paths`, `has_only_portable_paths`, and `has_no_randomness`, cleared the log file for the provided directory and rendered the project before executing the rest of the function. 

Sandboxing the provided path does not appear to work because if files are referenced outside the directory given (for example, "../data/data.csv"), they will not be found relative to the new sandbox directory. Instead, I found it worked best to clear the log and render from the original provided directory, rather than use a sandbox.

`has_no_absolute_paths` and `has_only_portable_paths` checks now work. `has_no_randomness` does NOT, which appears to likely be a problem within the writing of the check itself.

beanumber commented:
Hmm...

1. Will the paths not be found because they are absolute? I suppose that would be a problem...
2. Are there circumstances in which we wouldn't want to clear someone's log file?

ambertin commented:
1. I think the issue is that the sandbox only moves files at the same level or lower in the project hierarchy than the directory you give to the function. So anything higher up becomes no longer accessible. 

2. That’s something I was wondering about as well. Assuming you don’t delete the log, if someone has made changes to a file (perhaps changed some paths from absolute to relative), but their log still had paths from before they made changes, the path checks might fail, when they really shouldn’t. In that case, you would need to clear the log to get an accurate result. 

One solution to this problem of giving an option of whether or not to delete the log might be to add a TRUE/FALSE parameter to the function call where you can specify whether or not you’d like to delete the log file.

ambertin commented:
• `log_touch()` now takes a directory. Since `log_touch()` is used in most log functions, those others now take directories as well.
• `path_render_log()` removed
• Added environment variable `FERTILE_RENDER_MODE`. `path_log()` now returns the path to the interactive log when render mode is FALSE and the path to the render log when render mode is TRUE. 
• Functions that now use `FERTILE_RENDER_MODE` to access the render log are:
1. `proj_render()`
2. `has_rendered()`
3. `has_no_absolute_paths()`
4. `has_only_portable_paths()`
5. `has_no_randomness()` *note that this still does not work correctly
6. `proj_analyze_paths()`

• `has_rendered()` now looks at the last modified date of the new render log

ambertin commented:
Added a log push of "proj_render" to the render log at the end of `proj_render`. This fixes the problem of functions crashing when trying to access an empty render log, since the render log will always contain a line saying `proj_render` was called.

Added a `has_rendered()` check to proj_test so it only renders if the directory has been updated since the last render timestamp.
